### ISSUE TITLE
extra_js_urls and extra_css_urls should respect base_url setting

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#1041]
_Originally posted by @simonw in https://github.com/simonw/datasette/issues/1033#issuecomment-714681365_

Refs #1023

**Comment by simonw** [comment]
@psychemedia said: https://github.com/simonw/datasette/issues/1033#issuecomment-714657366

> How does `/-/static` relate to [current guidance docs around `static`](https://docs.datasette.io/en/latest/custom_templates.html?highlight=static#serving-static-files) regarding the `--static option` and metadata formulations such as `"extra_js_urls": [ "/static/app.js"]` (I've not managed to get this to work in a Jupyter server proxied set up; the [datasette / jupyter server proxy repo](https://github.com/simonw/jupyterserverproxy-datasette-demo) may provide a useful test example, eg via MyBinder, for folk to crib from?)

**Comment by simonw** [comment]
I'll need to update these docs once there's a solution for this in place: https://docs.datasette.io/en/latest/custom_templates.html#serving-static-files

**Comment by simonw** [comment]
I think I'll do this by looking for URLs that start with `/` - since it's also possible to have full `https://...` URLs in that setting.
```json
{
    "extra_css_urls": [
        "/static/styles.css"
    ],
    "extra_js_urls": [
        "/static/app.js"
    ]
}
```
I need to think about the `extra_css_urls` and `extra_js_urls` plugin hooks too: https://docs.datasette.io/en/stable/plugin_hooks.html#extra-css-urls-template-database-table-columns-view-name-request-datasette

**Comment by simonw** [comment]
Implemented in a4ca26a2659d21779adf625183061d8879954c15
